src/Microsoft.Health.Fhir.R4.Api/Features/Resources/Bundle/BundleHandler.cs

```diff
@@ -14,16 +14,16 @@ namespace Microsoft.Health.Fhir.Api.Features.Resources.Bundle
     /// </summary>
     public partial class BundleHandler
     {
-        private Dictionary<Hl7.Fhir.Model.Bundle.HTTPVerb, List<RouteContext>> GenerateRequestDictionary()
+        private Dictionary<Hl7.Fhir.Model.Bundle.HTTPVerb, List<(RouteContext, int)>> GenerateRequestDictionary()
         {
-            return new Dictionary<Hl7.Fhir.Model.Bundle.HTTPVerb, List<RouteContext>>
+            return new Dictionary<Hl7.Fhir.Model.Bundle.HTTPVerb, List<(RouteContext, int)>>
             {
-                { Hl7.Fhir.Model.Bundle.HTTPVerb.DELETE, new List<RouteContext>() },
-                { Hl7.Fhir.Model.Bundle.HTTPVerb.GET, new List<RouteContext>() },
-                { Hl7.Fhir.Model.Bundle.HTTPVerb.HEAD, new List<RouteContext>() },
-                { Hl7.Fhir.Model.Bundle.HTTPVerb.PATCH, new List<RouteContext>() },
-                { Hl7.Fhir.Model.Bundle.HTTPVerb.POST, new List<RouteContext>() },
-                { Hl7.Fhir.Model.Bundle.HTTPVerb.PUT, new List<RouteContext>() },
+                { Hl7.Fhir.Model.Bundle.HTTPVerb.DELETE, new List<(RouteContext, int)>() },
```

Comment:
Not your code, but I would like to suggest a small:

What we care about is the order in which the operations gets executed (and GenerateRequestDictionary really is just operating on top of that) so, I think we can simplify the code to:

In BundleHandler.cs, define a member variable like

`private readonly Hl7.Fhir.Model.Bundle.HTTPVerb[] _verbExecutionOrder;`

In shared BundleHandler class, add `:this()` after the constructor.

Then in the Stu3 and R4 specific BundleHandler class, add the private constructor that initializes version specific orders:

Stud3:
```
        private BundleHandler()
        {
            _verbExecutionOrder = new []
            {
                Hl7.Fhir.Model.Bundle.HTTPVerb.DELETE,
                Hl7.Fhir.Model.Bundle.HTTPVerb.POST,
                Hl7.Fhir.Model.Bundle.HTTPVerb.PUT,
                Hl7.Fhir.Model.Bundle.HTTPVerb.GET,
            };
        }
```

R4:
```
        private BundleHandler()
        {
            _verbExecutionOrder = new []
            {
                Hl7.Fhir.Model.Bundle.HTTPVerb.DELETE,
                Hl7.Fhir.Model.Bundle.HTTPVerb.POST,
                Hl7.Fhir.Model.Bundle.HTTPVerb.PUT,
                Hl7.Fhir.Model.Bundle.HTTPVerb.PATCH,
                Hl7.Fhir.Model.Bundle.HTTPVerb.GET,
                Hl7.Fhir.Model.Bundle.HTTPVerb.HEAD,
            };
        }
```
Remove both GenerateRequestDictionary and ExecuteAllRequests from version specific BundleHandler.cs files and move the logic into the shared BundleHandler.cs

Reply:
👍